osu.Game/Database/LegacyBeatmapExporter.cs

```diff
@@ -1,18 +1,91 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
 using osu.Framework.Platform;
 using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.Extensions;
+using osu.Game.IO;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Skinning;
+using osuTK;
 
 namespace osu.Game.Database
 {
+    /// <summary>
+    /// Exporter for osu!stable legacy beatmap archives.
+    /// Converts all beatmaps in the set to legacy format and exports it as a legacy package.
+    /// </summary>
     public class LegacyBeatmapExporter : LegacyArchiveExporter<BeatmapSetInfo>
     {
         public LegacyBeatmapExporter(Storage storage)
             : base(storage)
         {
         }
 
+        protected override Stream? GetFileContents(BeatmapSetInfo model, INamedFileUsage file)
+        {
+            bool isBeatmap = model.Beatmaps.Any(o => o.Hash == file.File.Hash);
+
+            if (!isBeatmap)
+                return base.GetFileContents(model, file);
+
+            // Read the beatmap contents and skin
+            using var contentStream = UserFileStorage.GetStream(file.File.GetStoragePath());
```

Comment:
Could call `base.GetFileContents(model, file)` instead? (Same applies lower down, too)

Reply:
yeah that reduces the code duplication a little bit